app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java

```diff
@@ -504,188 +509,172 @@ public void onClick(DialogInterface dialog, int which)
                 
             }
         });
-        findViewById(R.id.btnRotateLeftRight).setOnClickListener(
-				new View.OnClickListener() {
-				    public void onClick(View v) 
-				    {					    	
-				    	try{
-				    		/*doRotate(mImageView, fromDegree, toDegree);
-							fromDegree = (toDegree == 360) ? 0 : toDegree;
-							toDegree += 90;
-							if (toDegree > 360) {
-								toDegree = 90;
-							}*/
-				    		doRotate90(mImageView);
-				    		mImageView.mHighlightViews.clear();
-				    		mBitmap.recycle();
-				    		mBitmap = modifiedBitmap;
-				    		startFaceDetection();
-							
-				    	}catch(OutOfMemoryError o){
-				    		o.printStackTrace();
-				    	}
-				    	
-				    }
-				});
+
+        findViewById(R.id.btnRotateLeftRight).setOnClickListener (
+                new View.OnClickListener() {
+                    public void onClick (View v) {
+                        try {
+                            /*doRotate(mImageView, fromDegree, toDegree);
+                            fromDegree = (toDegree == 360) ? 0 : toDegree;
+                            toDegree += 90;
+                            if (toDegree > 360) {
+                                toDegree = 90;
+                            }*/
+                            // Flags an edit
+                            if (turns % 3 == 0 && turns != 0) {
```

Comment:
There is a bug. If you apply a filter first and then rotate image 3 times and save. It will save only one copy. Try it.